meson.build

```diff
@@ -0,0 +1,110 @@
+project(
+  'flatpak-builder',
+  'c',
+  license: 'LGPL-2.1-or-later',
+  meson_version: '>=0.49',
+  version: '1.0.11',
+  default_options: '-Dc_std=gnu99',
+)
+
+# TODO: do some cleaning up of useless args that meson already sets by default
+add_project_arguments(
+  '-Werror=empty-body',
+  '-Werror=strict-prototypes',
+  '-Werror=missing-prototypes',
+  '-Werror=implicit-function-declaration',
+  '-Werror=format=2',
+  '-Werror=format-security',
+  '-Werror=format-nonliteral',
+  '-Werror=pointer-arith',
+  '-Werror=init-self',
+  '-Werror=missing-declarations',
+  '-Werror=return-type',
+  '-Werror=overflow',
+  '-Werror=int-conversion',
+  '-Werror=parentheses',
+  '-Werror=incompatible-pointer-types',
+  '-Werror=misleading-indentation',
+  '-Werror=missing-include-dirs',
+  # Necessary for non POSIX stuff in glibc like fallocate
+  '-D_GNU_SOURCE',
+  language: 'c',
+)
+
+inc_dirs = include_directories(
+  '.',
+# FIXME: handle $prefixes/libdwarf/ too or let it to the option? The later would be easier
+  # Dwarf header stuff
+  get_option('dwarf_header_dir'),
+)
+
+cc = meson.get_compiler('c')
+
+# FIXME: this was in the autoconf file but it didn't prevent from configuring. Also, it seems like fdwalk isn't called in the code?!
+# cc.has_function('fdwalk')
+if not cc.has_header('sys/xattr.h')
+  error('You must have the sys/xattr.h header from glibc')
+endif
+if not cc.has_header('sys/capability.h')
+  error('sys/capability header not found. You need to install the libcap headers')
+endif
+
+if not cc.has_header('dwarf.h', include_directories: inc_dirs)
+  error('dwarf.h header could not be found in either the include directories or @0@'
+    .format(get_option('dwarf_header_dir'))
+  )
+endif
+
+libglnx_dep = subproject('libglnx').get_variable('libglnx_dep')
+libyaml_dep = dependency('yaml-0.1', required: get_option('yaml'))
+libelf_dep = dependency('libelf', version: '>=0.8.12')
```

Comment:
If I remember correctly, libelf has not always had pkg-config metadata - but hopefully all Linux distros are now using `elfutils`, which does have it. See a recent GLib version's build system, specifically `gio/meson.build`.

GLib's `gio/meson.build` suggests that the fallback to `cc.find_library` and `cc.has_function` is only necessary on \*BSD. If that's true, then we don't (need to) care about that case for Flatpak, because Flatpak is Linux-specific and is not portable to \*BSD anyway.

Reply:
Nice, I was surprised this manual header check was needed, so I'll remove it :)